src/framework/filterControl/forms/MixedTypeValueFilterForm.jsx

```diff
@@ -0,0 +1,159 @@
+
+import PropTypes from 'prop-types';
+import React, {
+  Component,
+} from 'react';
+
+import {
+  Filter,
+  FilterOption,
+} from '../../services';
+
+import CheckBox from '../../checkBox/CheckBox.jsx';
+import MixedTypeValueFilter from '../../services/filter/MixedTypeValueFilter';
+import PrimaryButton from '../../button/PrimaryButton.jsx';
+
+export default class MixedTypeValueFilterForm extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.onCheckBoxClick = this.onCheckBoxClick.bind(this);
+    this.onClickAddButton = this.onClickAddButton.bind(this);
+    this.onChange = this.onChange.bind(this);
+
+    this.state = {
+      ...this.mapComparisonValuetoState(props),
+      ...this.mapOptionsToState(props),
+    };
+  }
+
+  componentDidMount() {
+    // eslint-disable-next-line react/no-did-mount-set-state
+    this.setState({
+      ...this.mapComparisonValuetoState(this.props),
+      ...this.mapOptionsToState(this.props),
+    });
+  }
+
+  componentDidUpdate(prevProps) {
+    if (
+      !this.deepEqualsProps(prevProps.filterOption, this.props.filterOption) ||
```

Comment:
At this point this just compares two serealisable objects, so can we just call it "deepEqual"?